### ISSUE TITLE
AvailableTerritories is deprecated and fastlane is not upgraded to new api endpoints

### ISSUE THREAD

**Issue by shernade** [fastlane/fastlane#21968]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
There're some deprecated apple endpoints in fastlane: https://developer.apple.com/documentation/appstoreconnectapi/app_store_connect_api_release_notes/app_store_connect_api_2_3_release_notes#4165225

I was not able to check all of them but regarding https://developer.apple.com/documentation/appstoreconnectapi/list_all_available_territories_for_an_app
we should be using this endpoint:
GET https://api.appstoreconnect.apple.com/v1/appAvailabilities/{id}/availableTerritories
and NOT this:
GET https://api.appstoreconnect.apple.com/v1/apps/{id}/availableTerritories

Do we have a plan to start using new endpoints?

Thanks a lot

**Comment by lucgrabowski** [comment]
Hi, there is a PR https://github.com/fastlane/fastlane/pull/21901 with an implementation for request to get app availabilities, please take a look at the description there. For now it's left as a draft as there is one PR (https://github.com/fastlane/fastlane/pull/21900) that should be finished in the first place.

**Comment by jakub-przy** [comment]
Hey 👋 
Have we got any ETA on the above? 

Thanks,

**Comment by pedro380085** [comment]
This is fantastic, shall fix all the `price_tier` and `App has no prices yet... Enabling all countries in App Store Connect` errors we are seeing with Deliver.

**Comment by assembleMHN** [comment]
I hope this can be added soon. We work with a white label app, and im creating 5-10 new apps on new accounts every week, and its very annoying that this cannot be done. I had it scripted that fastlane would do it all, but now it crashes everytime it has to submit a new created app for review. So i have to go in manually and set the territory and pricing and then run fastlane to submit again. It has been like this for a very long time now.

**Comment by shernade** [comment]
i have this PR in my fork https://github.com/shernade/fastlane/pull/1

**Comment by muhammad67uzair** [comment]
Im trying to set availability for my apps using latest fastlane version 2.221.1.
but i keep getting this error:
```
[!] The provided entity includes an unknown relationship - 'availableTerritories' is not a relationship on the resource 'apps' - /data/relationships/availableTerritories (Spaceship::UnexpectedResponse)
```

i tried using other versions such as 2.213.0, 2.219.0, 2.218.0, but it wont budge.

is it possible that the PRs mentioned above will fix this? if so, when will this be rolled out?

**Comment by assembleMHN** [comment]
Please, fastlane has been broken with pricing and avalibility for sooo long now. this only happens when creating new apps (which we do alot) and its annoying to have a fully automated system for everything app related but still have to go in and set pricing and avalibility manually everytime. Hope that this will be solved soon, its been an issue for over 1 year by now :(

**Comment by Goles** [comment]
@assembleMHN  Does Fastlane work for _updating_ pricing and availability for existing apps though?

**Comment by jakub-przy** [comment]
> @assembleMHN Does Fastlane work for _updating_ pricing and availability for existing apps though?

Yeah, that seems to be fine @Goles, just used that myself

**Comment by moazelsawaf** [comment]
Any updates?

**Comment by pusswzy** [comment]
Still waiting this fix, it's important to me

**Comment by ScottBishop** [comment]
Still waiting on this fix as well

**Comment by assembleMHN** [comment]
> @assembleMHN Does Fastlane work for _updating_ pricing and availability for existing apps though?

No it does not, but the issue only shows when creating apps because the automatic push to review (after creating) failes when pricing is not set and you have to do it manually. After that it works fine because you no longer set the pricing (as its already set). But updating pricing through fastlane dont work after they changed the way its done almost a year ago. But not enough people are using this feature because its 99.9% of the time its only done onces with app creation so there is not alot of noise around this broken/missing featuire :(

**Comment by assembleMHN** [comment]
So im also still waiting for a fix!

**Comment by mobligo** [comment]
still waiting for the fix

**Comment by Codewealth** [comment]
Can we please get a fix for this?

**Comment by Codewealth** [comment]
> I hope this can be added soon. We work with a white label app, and im creating 5-10 new apps on new accounts every week, and its very annoying that this cannot be done. I had it scripted that fastlane would do it all, but now it crashes everytime it has to submit a new created app for review. So i have to go in manually and set the territory and pricing and then run fastlane to submit again. It has been like this for a very long time now.

It's so annoying. Not sure why nobody is fixing this.

**Comment by Codewealth** [comment]
> Please, fastlane has been broken with pricing and avalibility for sooo long now. this only happens when creating new apps (which we do alot) and its annoying to have a fully automated system for everything app related but still have to go in and set pricing and avalibility manually everytime. Hope that this will be solved soon, its been an issue for over 1 year by now :(

Yeah, it's terrible. Same here. We deploy like 10+ apps every day and this breaks the entire automation flow. It's quite frustrating.

**Comment by qfrewin-iclasspro** [comment]
I'm another white label app creator.  I'd love it if we could get this fix soon.

**Comment by assembleMHN** [comment]
Please could we get an update or fix for this, when the automation tool cannot automate its annoying and this is an issue for us eveyday. Hope somebody can look at this :)

**Comment by rasheedk** [comment]
Simply fixed it for new app in a way that first submit build for review and immediately then update AvailableTerritories . it should work.

**Comment by iBotPeaches** [comment]
Best I understand reading this thread. Spaceship gained access to the new endpoints, but nothing yet has happened to swap implementations to it?

Ping on me PR if anyone can do this.

**Comment by Blackjacx** [comment]
Here's a working workaround which directly calls the ASC API:

```ruby
private_lane :update_app_availability_v2 do |options|
  app_id = options[:app_id]
  available_territory_ids = options[:available_territory_ids]

  client = Spaceship::ConnectAPI.client.tunes_request_client
  params = client.build_params(
    filter: {},
    includes: "territory",
    limit: 200,
    sort: nil
  )
  response = client.get("v2/appAvailabilities/#{app_id}/territoryAvailabilities", params).body
  territory_availabilities = response["data"] || []

  UI.user_error!("No territory availabilities found for app ID #{app_id}") if territory_availabilities.empty?

  updates = []

  territory_availabilities.each do |territory_availability|
    territory_id = territory_availability.dig("relationships", "territory", "data", "id")
    next unless is_set(territory_id)

    should_be_available = available_territory_ids.include?(territory_id)
    currently_available = territory_availability.dig("attributes", "available")

    next if currently_available == should_be_available

    updates << {
      id: territory_availability["id"],
      available: should_be_available,
      territory_id: territory_id
    }
  end

  updates.each do |update|
    body = {
      data: {
        type: "territoryAvailabilities",
        id: update[:id],
        attributes: {
          available: update[:available]
        }
      }
    }

    client.patch("v1/territoryAvailabilities/#{update[:id]}", body)
  end

  UI.message("Updated #{updates.count} territory availabilities for app #{app_id}")
end
```


Just call it like this and feed it the 3-letter country IDs where your app should be available:

```ruby
  update_app_availability_v2(
    app_id: app.id,
    available_territory_ids: available_territory_ids
  )
```

> [!note]
> I really don't want this code in my Fastfile but it is necessary since this thicket here is not solved yet. So any news on the progress here?

**Comment by iBotPeaches** [comment]
Sorry for delay. Do others believe this PR fixes this? https://github.com/fastlane/fastlane/pull/29650 it swaps the usage to the v2 endpoint

**Comment by iBotPeaches** [comment]
I dug into this a bit. My concern so looking for feedback from others is this endpoint I don't see called anywhere in fastlane codebase, so purely a Spaceship API afaik.

The old endpoint (v1) gave you only territories the app is in. It was pretty simple since it was returned its in use. This v2 endpoint is different, it shows ALL territories with a status of like (available, cannot_sell, available_for_sale, etc)

So this PR swapping the endpoint over - https://github.com/fastlane/fastlane/pull/29650, has a completely different return than it used to be. I have no experience how we handle a change of a v1 -> v2 endpoint Spaceship and what this means when the return is completely different.
